lib/ansible/modules/cloud/misc/virt_secret.py

```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Theo Ouzhinski <touzhinski@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+    module: virt_secret
+    short_description: Manages Libvirt secrets
+    description:
+        - Manages Libvirt secrets.
+    version_added: "2.8"
+    author: "Theo Ouzhinski (@theo-o)"
+    options:
+        uuid:
+            description:
+                - The UUID of the secret being managed.
+                - Must be in the format specified by RFC4122.
+                - Either this or I(secret) must be used to identify the secret in question.
+        secret:
+            description:
+                - A Base64-encoded secret value.
+                - Either this or I(uuid) must be used to identify the secret in question.
+        usage:
+            description:
+                - Describes what the secret is used for.
+                - Required when I(state=present).
+            choices:
+                - volume
+                - ceph
+                - iscsi
+                - tls
+        usage_element:
+            description:
+                - Required when usage is defined.
+                - See L(the Libvirt docs, https://libvirt.org/formatsecret.html) for more information.
```

Comment:
I read the docs page, but it's not at all clear what this usage_element is for, nor do the examples provided below give me a good idea.